osgbuild/fetch_sources.py

```diff
@@ -169,6 +169,11 @@ def process_meta_url(line, destdir, nocheck):
 
     return files
 
+def deref_git_sha(sha):
+    output, rc = utils.sbacktick(["git", "rev-parse", sha + "^{}"])
```

Comment:
Not like it's likely to come up, but would this fail (exit nonzero) if the sha was, say, a root commit?
Also, the output might need to be stripped.

Reply:
> Not like it's likely to come up, but would this fail (exit nonzero) if the sha was, say, a root commit?

No - if the sha is a commit, it will just return that commit; and if the sha is a tag object, it will dereference until it finds a non-tag object.  From `git help rev-parse`:

```
       <rev>^{}, e.g. v0.99.8^{}
           A suffix ^ followed by an empty brace pair means the object could
           be a tag, and dereference the tag recursively until a non-tag
           object is found.
```

Anyway in this case there's nothing special about root commits.

> Also, the output might need to be stripped.

I specifically checked for this and found that `utils.sbacktick()` returns a `.strip()`ed copy, unless you pass `nostrip=True`, or something like that.

Reply:
Re: stripping in `utils.sbacktick()`:
https://github.com/opensciencegrid/osg-build/blob/master/osgbuild/utils.py#L113-L139

Reply:
I'll take your word for it :)

Can you try a scratch build to see if it works (or at least doesn't cause a regression?)

Reply:
> scratch build plz

Alright, i kicked off (and cancelled) 4 scratch builds to test different cases:
1. `scitokens` with `hash=d6cfc27ae40ba1d272dfc69ee67f18480630b292` (the sha for the heavy-weight `tag`)
2. `scitokens` with `hash=025cda070a922fad40f0b2384fd72a8498805407` (the sha for the `commit` object)
3. `scitokens` with `hash=025cda070a922fad40f0b2384fd72a8498805408` (a non-existant sha)
4. `osg-build` with its current `hash` (which is a sha for a `commit` object)

1,2, & 4 succeeded as expected; 3 failed (as expected), with the following output on stderr:

```
fatal: ambiguous argument '025cda070a922fad40f0b2384fd72a8498805408^{}': unknown revision or path not in the working tree.
Use '--' to separate paths from revisions, like this:
'git <command> [<revision>...] -- [<file>...]'
-------------------------------------------------------------------------------
Git failed to parse rev: '025cda070a922fad40f0b2384fd72a8498805408'
-------------------------------------------------------------------------------
```

So it looks to be working as expected.  I think that "revision or path not in the working tree" message is useful, but unfortunately git outputs those next two lines of noise on errors even if i include a `--` argument in the right place.

Reply:
Unfortunate but acceptable.